src/imp.rs

```diff
@@ -644,33 +728,49 @@ fn sign_stct<Rng: RngCore + CryptoRng>(
 fn stake_sign(
     sk: &SecretKey,
     pk: &PublicKey,
+    counter: u64,
     value: u64,
-    created_at: u64,
 ) -> Signature {
-    let mut bytes = Vec::with_capacity(16);
+    let mut msg = Vec::with_capacity(u64::SIZE + u64::SIZE);
+
+    msg.extend(counter.to_bytes());
+    msg.extend(value.to_bytes());
+
+    sk.sign(pk, &msg)
+}
+
+/// Creates a signature compatible with what the stake contract expects for a
+/// unstake transaction.
+fn unstake_sign(
+    sk: &SecretKey,
+    pk: &PublicKey,
+    counter: u64,
+    note: Note,
+) -> Signature {
+    let mut msg = Vec::with_capacity(u64::SIZE + Note::SIZE);
 
-    bytes.extend(value.to_bytes());
-    bytes.extend(created_at.to_le_bytes());
+    msg.extend(counter.to_bytes());
+    msg.extend(note.to_bytes());
 
-    sk.sign(pk, &bytes)
+    sk.sign(pk, &msg)
 }
 
 /// Creates a signature compatible with what the stake contract expects for a
 /// withdraw transaction.
 fn withdraw_sign(
     sk: &SecretKey,
     pk: &PublicKey,
-    stake: &StakeInfo,
-    note: &Note,
+    counter: u64,
```

Comment:
What the `counter` is for? It's the same as `created_at`? Why the change?

Reply:
The change is described in more detail in https://github.com/dusk-network/rusk/pull/698, but the gist of it is that if we're using `created_at` as a counter - we might as well call it a `counter`.
There was also some tricky block height logic and this allows us to remove that here in the stake contract and, by extension, here as well.

Reply:
Maybe add a comment explaining what should be in this argument? Its not immediately obvious.